promtimer/promtimer.py

```diff
@@ -375,6 +379,22 @@ def main():
         times = cbstats.CBCollect.compute_min_and_max_times(stats_sources)
         min_time = datetime.datetime.fromtimestamp(times[0]).isoformat()
         max_time = datetime.datetime.fromtimestamp(times[1]).isoformat()
+
+        timezone_override = os.environ.get('GRAFANA_TZ', '')
+        if timezone_override != '':
+            default_timezone = timezone_override
+        else:
+            try:
+                default_timezone = next(
+                    map(lambda s: s.get_timezone(), stats_sources), None)
+                if default_timezone is None:
+                    default_timezone = 'Etc/UTC'
+            except Exception as e:
+                logging.error(
+                    'failed to determine the cluster timezone: {}'.format(e))
```

Comment:
I think it would be nice to create a static method in `CBCollect` that will take a lists of CBCollect stats sources and does the parsing and exception catching and returns to the caller here in promtimer.

Reply:
Done